packages/sdk-core/src/BacktraceCoreClient.ts

```diff
@@ -131,25 +143,27 @@ export abstract class BacktraceCoreClient {
 
         this.attributeManager = new AttributeManager(attributeProviders);
 
+        const stackTraceConverter = this._setup.stackTraceConverter ?? new V8StackTraceConverter();
         this._dataBuilder = new BacktraceDataBuilder(
             this._sdkOptions,
             stackTraceConverter,
             this.attributeManager,
             new DebugIdProvider(stackTraceConverter, this._setup.debugIdMapProvider),
         );
 
-        this._attachments = this.options.attachments ?? [];
+        if (this.options?.database?.enable === true) {
+            const provider =
+                this._setup.databaseStorageProvider ??
```

Comment:
You can drop databaseStorageProvider support and use only a new file system. The core client interface is not exposed outside the package and also to construct it you need to have package internals not exposed anywhere else.

Reply:
I'm not sure what you mean? Use `fileSystem` directly in `BacktraceDatabase`?

Reply:
I think you're trying to use databaseStorageProvider - I think you can remove it and use only the new file system variable

Reply:
so we will not be planning to override providers from outside, that's what you mean?

Reply:
is there any reason to do that if we have file system? I think, no

Reply:
Ok, removed